addons/web_unsplash/static/src/services/unsplash_service.js

```diff
@@ -7,34 +7,36 @@ export const unsplashService = {
     async start(env, { rpc }) {
         const _cache = {};
         return {
-            async getImages(query, offset = 0, pageSize = 30) {
+            async getImages(query, offset = 0, pageSize = 30, landscape) {
                 const from = offset;
                 const to = offset + pageSize;
-                let cachedData = _cache[query];
+                let cachedData = _cache[query + landscape];
 
                 if (cachedData && (cachedData.images.length >= to || (cachedData.totalImages !== 0 && cachedData.totalImages < to))) {
                     return { images: cachedData.images.slice(from, to), isMaxed: to > cachedData.totalImages };
                 }
-                cachedData = await this._fetchImages(query);
+                cachedData = await this._fetchImages(query, landscape);
                 return { images: cachedData.images.slice(from, to), isMaxed: to > cachedData.totalImages };
             },
             /**
              * Fetches images from unsplash and stores it in cache
              */
-            async _fetchImages(query) {
-                if (!_cache[query]) {
-                    _cache[query] = {
+            async _fetchImages(query, landscape) {
+                const key = query + landscape;
```

Comment:
Without the orientation in the cache key, we won't fetch new images with the specified orientation if we use the same query term.

Eg: search for "cat" images in the article body, then search for "cat" images when selecting a cover. Since the query word is in the cache, you would get the images fetched for the article body, hence images that are not necessarily in the landscape orientation.

Reply:
(Probably needs a comment in the code then).